doc/developer-guide.md

```diff
@@ -0,0 +1,260 @@
+# ddf-validation Developer's guide 
+
+## Registry
```

Comment:
What is registry (for me it is clear) but I think it should be explained in the doc.